### PR TITLE
Use Mustache.java for template rendering

### PR DESCRIPTION
**Pull request by murdos**
This brings possibility to do more advanced templating, rather than just arguments replacement.

Use this enhanced templating to simplify generated `SpringLiquibaseUtilTest`

### CODE REVIEW

**Comment by murdos** [comment]
@swarajsaaj: If you want to take a look, since you work on #9332 inspired me for these changes. I don't know why, but I can't add you to reviewers :/

**Comment by swarajsaaj** [comment]
> @swarajsaaj: If you want to take a look, since you work on #9332 inspired me for these changes. I don't know why, but I can't add you to reviewers :/

Awesome work @murdos 👍 , `SpringLiquibaseUtilTest` looks much better now, the way it should be.

File: src/main/resources/generator/server/springboot/dbmigration/liquibase/test/SpringLiquibaseUtilTest.java.mustache [hunk 1]
```diff
@@ -219,22 +222,25 @@ class SpringLiquibaseUtilTest {
...
-        properties = yaml.getObject();
-      } else {
-        properties = new Properties();
-        properties.load(
-        SpringLiquibaseUtilTest.class.getClassLoader().getResourceAsStream("config/application-test.properties"));
-      }
+          SpringLiquibaseUtilTest.class.getClassLoader().getResource("config/application-test.yml").toURI())
+        );
+        return yaml.getObject();
     } catch (IOException | URISyntaxException exception) {
```
Comment (swarajsaaj) [inline src/main/resources/generator/server/springboot/dbmigration/liquibase/test/SpringLiquibaseUtilTest.java.mustache:233]: Just URISyntaxException here?

Comment (swarajsaaj) [inline src/main/resources/generator/server/springboot/dbmigration/liquibase/test/SpringLiquibaseUtilTest.java.mustache:233]:
```suggestion
    } catch (URISyntaxException exception) {
```

Comment (murdos) [reply src/main/resources/generator/server/springboot/dbmigration/liquibase/test/SpringLiquibaseUtilTest.java.mustache]: Yes, should be fixed now :)

**Comment by pascalgrimaud** [comment]
Adding a bounty as it seems advanced work

**Comment by murdos** [comment]
Bounty claimed: https://opencollective.com/generator-jhipster/expenses/208286

**Comment by pascalgrimaud** [comment]
@murdos : approved

**Pull request merged**